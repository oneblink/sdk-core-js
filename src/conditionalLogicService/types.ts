import { FormTypes, SubmissionTypes } from '@oneblink/types'

export type FormElementsCtrl = {
  model: SubmissionTypes.S3SubmissionData['submission']
  flattenedElements: FormTypes.FormElement[]
  parentFormElementsCtrl?: FormElementsCtrl
}
