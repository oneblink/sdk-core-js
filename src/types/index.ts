export type FormElementsCtrl = {
  model: Record<string, unknown>
  flattenedElements: import('@oneblink/types').FormTypes.FormElement[]
  parentFormElementsCtrl?: FormElementsCtrl
}

export type FormElementsConditionallyShown = Record<
  string,
  FormElementConditionallyShown
>

export type FormElementConditionallyShown =
  | undefined
  | {
      type: 'formElement'
      isHidden: boolean
      options?: import('@oneblink/types').FormTypes.ChoiceElementOption[]
    }
  | {
      type: 'formElements'
      isHidden: boolean
      formElements: FormElementsConditionallyShown | undefined
    }
  | {
      type: 'repeatableSet'
      isHidden: boolean
      entries: Record<string, FormElementsConditionallyShown | undefined>
    }

