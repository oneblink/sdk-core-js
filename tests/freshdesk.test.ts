import {
  generateFreshdeskTicketFieldDefinitions,
  setFreshdeskFieldOptions,
} from '../src/freshdeskService'
import { FreshdeskTypes } from '@oneblink/types'
import freshdeskFieldFixture from './freshdeskFieldsFixture.json'

describe('generateFreshdeskTicketFieldDefinitions', () => {
  it('should return an empty array when given an empty input', () => {
    expect(generateFreshdeskTicketFieldDefinitions([])).toEqual([])
  })

  it('should map TEXT_SINGLE_LINE fields correctly', () => {
    const fields: FreshdeskTypes.FreshdeskField[] = [
      //@ts-expect-error Not filling in all properties
      {
        id: 1,
        name: 'Field1',
        type: 'custom_text',
        label_for_customers: 'Field 1',
        required_for_customers: true,
      },
    ]
    expect(generateFreshdeskTicketFieldDefinitions(fields)).toEqual([
      {
        displayName: 'Field 1',
        freshdeskFieldName: 'Field1',
        id: '1',
        isRequired: true,
        type: 'TEXT_SINGLE_LINE',
      },
    ])
  })

  it('should map TEXT_MULTI_LINE fields correctly', () => {
    const fields: FreshdeskTypes.FreshdeskField[] = [
      //@ts-expect-error Not filling in all properties
      {
        id: 2,
        name: 'Field2',
        type: 'custom_paragraph',
        label_for_customers: 'Field 2',
        required_for_customers: true,
      },
    ]
    expect(generateFreshdeskTicketFieldDefinitions(fields)).toEqual([
      {
        displayName: 'Field 2',
        freshdeskFieldName: 'Field2',
        id: '2',
        isRequired: true,
        type: 'TEXT_MULTI_LINE',
      },
    ])
  })

  it('should map CHOICE_SINGLE fields correctly', () => {
    const fields: FreshdeskTypes.FreshdeskField[] = [
      //@ts-expect-error Not filling in all properties
      {
        id: 3,
        name: 'Field3',
        type: 'custom_dropdown',
        label_for_customers: 'Field 3',
        required_for_customers: true,
        options: [
          { label: 'option 1', value: 'value 1' },
          { label: 'option 2', value: 'value 2' },
        ],
      },
    ]
    expect(generateFreshdeskTicketFieldDefinitions(fields)).toEqual([
      {
        checkIsFormElementSupported: expect.any(Function),
        choices: [
          {
            label: 'option 1',
            value: 'value 1',
          },
          {
            label: 'option 2',
            value: 'value 2',
          },
        ],
        displayName: 'Field 3',
        freshdeskFieldName: 'Field3',
        id: '3',
        isRequired: true,
        type: 'CHOICE_SINGLE',
      },
    ])
  })

  it('should map DATE fields correctly', () => {
    const fields: FreshdeskTypes.FreshdeskField[] = [
      //@ts-expect-error Not filling in all properties
      {
        id: 4,
        name: 'Field4',
        type: 'custom_date',
        label_for_customers: 'Field 4',
        required_for_customers: true,
      },
    ]
    expect(generateFreshdeskTicketFieldDefinitions(fields)).toEqual([
      {
        displayName: 'Field 4',
        freshdeskFieldName: 'Field4',
        id: '4',
        isRequired: true,
        type: 'DATE',
      },
    ])
  })

  it('should map NUMBER fields correctly', () => {
    const fields: FreshdeskTypes.FreshdeskField[] = [
      //@ts-expect-error Not filling in all properties
      {
        id: 5,
        name: 'Field5',
        type: 'custom_number',
        label_for_customers: 'Field 5',
        required_for_customers: true,
      },
    ]
    expect(generateFreshdeskTicketFieldDefinitions(fields)).toEqual([
      {
        displayName: 'Field 5',
        freshdeskFieldName: 'Field5',
        id: '5',
        isRequired: true,
        type: 'NUMBER',
      },
    ])
  })

  it('should map BOOLEAN fields correctly', () => {
    const fields: FreshdeskTypes.FreshdeskField[] = [
      //@ts-expect-error Not filling in all properties
      {
        id: 6,
        name: 'Field6',
        type: 'custom_checkbox',
        label_for_customers: 'Field 6',
        required_for_customers: true,
      },
    ]
    expect(generateFreshdeskTicketFieldDefinitions(fields)).toEqual([
      {
        displayName: 'Field 6',
        freshdeskFieldName: 'Field6',
        id: '6',
        isRequired: true,
        type: 'BOOLEAN',
      },
    ])
  })

  it('should map EMAIL fields correctly', () => {
    const fields: FreshdeskTypes.FreshdeskField[] = [
      //@ts-expect-error Not filling in all properties
      {
        id: 7,
        name: 'Field7',
        type: 'default_requester',
        label_for_customers: 'Field 7',
        required_for_customers: true,
      },
    ]
    expect(generateFreshdeskTicketFieldDefinitions(fields)).toEqual([
      {
        displayName: 'Field 7',
        freshdeskFieldName: 'Field7',
        id: '7',
        isRequired: true,
        type: 'EMAIL',
      },
    ])
  })

  it('should map DEPENDENT_FIELD fields correctly', () => {
    const fields: FreshdeskTypes.FreshdeskField[] = [
      //@ts-expect-error Not filling in all properties
      {
        id: 8,
        name: 'Field8',
        type: 'nested_field',
        label_for_customers: 'Field 8',
        required_for_customers: true,
        options: [
          {
            label: 'AUS',
            value: 'AUS',
            options: [
              {
                label: 'NSW',
                value: 'NSW',
                options: [
                  {
                    label: 'Newcastle',
                    value: 'Newcastle',
                  },
                  {
                    label: 'Gosford',
                    value: 'Gosford',
                  },
                ],
              },
            ],
          },
        ],
      },
    ]
    expect(generateFreshdeskTicketFieldDefinitions(fields)).toEqual([
      {
        choices: [
          {
            label: 'AUS',
            subCategories: [
              {
                items: [
                  {
                    label: 'Newcastle',
                    value: 'Newcastle',
                  },
                  {
                    label: 'Gosford',
                    value: 'Gosford',
                  },
                ],
                label: 'NSW',
                value: 'NSW',
              },
            ],
            value: 'AUS',
          },
        ],
        displayName: 'Field 8',
        freshdeskFieldName: 'Field8',
        id: '8',
        isRequired: true,
        type: 'DEPENDENT_FIELD',
      },
    ])
  })

  it('should return the correct value when choices is used over options', () => {
    const fields: FreshdeskTypes.FreshdeskField[] = [
      //@ts-expect-error Not filling in all properties
      {
        id: 3,
        name: 'Field3',
        type: 'custom_dropdown',
        label_for_customers: 'Field 3',
        required_for_customers: true,
        choices: ['value 1', 'value 2'],
      },
    ]
    expect(generateFreshdeskTicketFieldDefinitions(fields)).toEqual([
      {
        checkIsFormElementSupported: expect.any(Function),
        choices: [
          {
            label: 'value 1',
            value: 'value 1',
          },
          {
            label: 'value 2',
            value: 'value 2',
          },
        ],
        displayName: 'Field 3',
        freshdeskFieldName: 'Field3',
        id: '3',
        isRequired: true,
        type: 'CHOICE_SINGLE',
      },
    ])
  })

  it('should map DEPENDENT_FIELD field correctly when choices is used', () => {
    const fields: FreshdeskTypes.FreshdeskField[] = [
      //@ts-expect-error Not filling in all properties
      {
        id: 8,
        name: 'Field8',
        type: 'nested_field',
        label_for_customers: 'Field 8',
        required_for_customers: true,
        choices: {
          AUS: {
            NSW: ['Sydney', 'Gosford'],
            QLD: ['Brisbane', 'Gold Coast'],
          },
          USA: {
            Texas: ['Austin', 'Dallas'],
            Kansas: ['Kansas City', 'Manhattan'],
          },
        },
      },
    ]
    expect(generateFreshdeskTicketFieldDefinitions(fields)).toMatchSnapshot()
  })
})

describe('setFreshdeskFieldOptions', () => {
  it('should transform all of the fields correctly', () => {
    const freshdeskFields = setFreshdeskFieldOptions(
      freshdeskFieldFixture as FreshdeskTypes.FreshdeskField[],
    )
    expect(freshdeskFields).toMatchSnapshot()

    expect(setFreshdeskFieldOptions(freshdeskFields)).toMatchSnapshot()
  })
})
