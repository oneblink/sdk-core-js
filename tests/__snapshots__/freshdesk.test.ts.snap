// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateFreshdeskTicketFieldDefinitions should map DEPENDENT_FIELD field correctly when choices is used 1`] = `
[
  {
    "choices": [
      {
        "label": "AUS",
        "subCategories": [
          {
            "items": [
              {
                "label": "Sydney",
                "value": "Sydney",
              },
              {
                "label": "Gosford",
                "value": "Gosford",
              },
            ],
            "label": "NSW",
            "value": "NSW",
          },
          {
            "items": [
              {
                "label": "Brisbane",
                "value": "Brisbane",
              },
              {
                "label": "Gold Coast",
                "value": "Gold Coast",
              },
            ],
            "label": "QLD",
            "value": "QLD",
          },
        ],
        "value": "AUS",
      },
      {
        "label": "USA",
        "subCategories": [
          {
            "items": [
              {
                "label": "Austin",
                "value": "Austin",
              },
              {
                "label": "Dallas",
                "value": "Dallas",
              },
            ],
            "label": "Texas",
            "value": "Texas",
          },
          {
            "items": [
              {
                "label": "Kansas City",
                "value": "Kansas City",
              },
              {
                "label": "Manhattan",
                "value": "Manhattan",
              },
            ],
            "label": "Kansas",
            "value": "Kansas",
          },
        ],
        "value": "USA",
      },
    ],
    "displayName": "Field 8",
    "freshdeskFieldName": "Field8",
    "id": "8",
    "isRequired": true,
    "type": "DEPENDENT_FIELD",
  },
]
`;

exports[`setFreshdeskFieldOptions should transform all of the fields correctly 1`] = `
[
  {
    "created_at": "2018-05-09T03:34:50Z",
    "customers_can_edit": true,
    "customers_can_filter": false,
    "default": true,
    "description": "Ticket requester",
    "displayed_to_customers": true,
    "id": 42000317068,
    "label": "Search a requester",
    "label_for_customers": "Email",
    "name": "requester",
    "portal_cc": true,
    "portal_cc_to": "all",
    "position": 1,
    "required_for_agents": true,
    "required_for_closure": false,
    "required_for_customers": true,
    "type": "default_requester",
    "updated_at": "2021-04-28T00:31:30Z",
  },
  {
    "created_at": "2018-05-09T03:34:50Z",
    "customers_can_edit": true,
    "customers_can_filter": false,
    "default": true,
    "description": "Ticket subject",
    "displayed_to_customers": true,
    "id": 42000317069,
    "label": "Subject",
    "label_for_customers": "Subject",
    "name": "subject",
    "position": 2,
    "required_for_agents": true,
    "required_for_closure": true,
    "required_for_customers": true,
    "type": "default_subject",
    "updated_at": "2020-08-03T04:22:47Z",
  },
  {
    "created_at": "2023-04-18T13:59:34Z",
    "customers_can_edit": false,
    "customers_can_filter": false,
    "default": false,
    "description": null,
    "displayed_to_customers": false,
    "id": 42000340772,
    "label": "Submission Id",
    "label_for_customers": "Submission Id",
    "name": "cf_submission_id",
    "position": 3,
    "required_for_agents": false,
    "required_for_closure": false,
    "required_for_customers": false,
    "type": "custom_text",
    "updated_at": "2023-04-19T02:36:46Z",
  },
  {
    "created_at": "2021-11-18T03:59:25Z",
    "customers_can_edit": true,
    "customers_can_filter": false,
    "default": false,
    "description": null,
    "displayed_to_customers": true,
    "id": 42000339370,
    "label": "Form ID",
    "label_for_customers": "Form ID",
    "name": "cf_form_id",
    "position": 4,
    "required_for_agents": false,
    "required_for_closure": false,
    "required_for_customers": false,
    "type": "custom_text",
    "updated_at": "2022-07-04T00:36:57Z",
  },
  {
    "choices": [
      "Customer",
      "CivicPlus",
      "OneBlink",
      "AWS",
      "Other",
    ],
    "created_at": "2020-08-03T04:39:49Z",
    "customers_can_edit": false,
    "customers_can_filter": false,
    "default": false,
    "description": null,
    "displayed_to_customers": false,
    "id": 42000337784,
    "label": "From",
    "label_for_customers": "From",
    "name": "cf_from",
    "options": [
      {
        "label": "Customer",
        "value": "Customer",
      },
      {
        "label": "CivicPlus",
        "value": "CivicPlus",
      },
      {
        "label": "OneBlink",
        "value": "OneBlink",
      },
      {
        "label": "AWS",
        "value": "AWS",
      },
      {
        "label": "Other",
        "value": "Other",
      },
    ],
    "position": 5,
    "required_for_agents": false,
    "required_for_closure": true,
    "required_for_customers": false,
    "type": "custom_dropdown",
    "updated_at": "2022-09-01T23:17:09Z",
  },
  {
    "created_at": "2018-05-09T03:34:51Z",
    "customers_can_edit": true,
    "customers_can_filter": false,
    "default": true,
    "description": "Ticket Company",
    "displayed_to_customers": true,
    "id": 42000317078,
    "label": "Company",
    "label_for_customers": "Company",
    "name": "company",
    "position": 6,
    "required_for_agents": true,
    "required_for_closure": true,
    "required_for_customers": true,
    "type": "default_company",
    "updated_at": "2020-08-03T04:22:59Z",
  },
  {
    "choices": [
      "Question",
      "Issue",
      "Feature Request",
      "Other",
    ],
    "created_at": "2018-05-09T03:34:50Z",
    "customers_can_edit": false,
    "customers_can_filter": false,
    "default": true,
    "description": "Ticket type",
    "displayed_to_customers": true,
    "id": 42000317070,
    "label": "Type",
    "label_for_customers": "Type of Request",
    "name": "ticket_type",
    "options": [
      {
        "label": "Question",
        "value": "Question",
      },
      {
        "label": "Issue",
        "value": "Issue",
      },
      {
        "label": "Feature Request",
        "value": "Feature Request",
      },
      {
        "label": "Other",
        "value": "Other",
      },
    ],
    "position": 7,
    "required_for_agents": true,
    "required_for_closure": true,
    "required_for_customers": false,
    "type": "default_ticket_type",
    "updated_at": "2022-12-07T22:16:24Z",
  },
  {
    "choices": {
      "Bot": 12,
      "Chat": 7,
      "Chat - Internal Task": 14,
      "Ecommerce": 11,
      "Email": 1,
      "Facebook": 6,
      "Feedback Widget": 9,
      "Forum": 4,
      "MobiHelp": 8,
      "Outbound Email": 10,
      "Phone": 3,
      "Portal": 2,
      "Twitter": 5,
      "Whatsapp": 13,
    },
    "created_at": "2018-05-09T03:34:50Z",
    "customers_can_edit": false,
    "customers_can_filter": false,
    "default": true,
    "description": "Ticket source",
    "displayed_to_customers": false,
    "id": 42000317071,
    "label": "Source",
    "label_for_customers": "Source",
    "name": "source",
    "options": [
      {
        "label": "Email",
        "value": 1,
      },
      {
        "label": "Portal",
        "value": 2,
      },
      {
        "label": "Phone",
        "value": 3,
      },
      {
        "label": "Forum",
        "value": 4,
      },
      {
        "label": "Twitter",
        "value": 5,
      },
      {
        "label": "Facebook",
        "value": 6,
      },
      {
        "label": "Chat",
        "value": 7,
      },
      {
        "label": "MobiHelp",
        "value": 8,
      },
      {
        "label": "Feedback Widget",
        "value": 9,
      },
      {
        "label": "Outbound Email",
        "value": 10,
      },
      {
        "label": "Ecommerce",
        "value": 11,
      },
      {
        "label": "Bot",
        "value": 12,
      },
      {
        "label": "Whatsapp",
        "value": 13,
      },
      {
        "label": "Chat - Internal Task",
        "value": 14,
      },
    ],
    "position": 8,
    "required_for_agents": false,
    "required_for_closure": false,
    "required_for_customers": false,
    "type": "default_source",
    "updated_at": "2022-05-08T23:59:38Z",
  },
  {
    "choices": {
      "10": [
        "Review",
        "In Review",
      ],
      "2": [
        "Open",
        "Being Processed",
      ],
      "3": [
        "Pending",
        "Awaiting your Reply",
      ],
      "4": [
        "Resolved",
        "This ticket has been Resolved",
      ],
      "5": [
        "Closed",
        "This ticket has been Closed",
      ],
      "9": [
        "In Backlog",
        "In Backlog",
      ],
    },
    "created_at": "2018-05-09T03:34:50Z",
    "customers_can_edit": false,
    "customers_can_filter": false,
    "default": true,
    "description": "Ticket status",
    "displayed_to_customers": true,
    "id": 42000317072,
    "label": "Status",
    "label_for_customers": "Status",
    "name": "status",
    "options": [
      {
        "label": "Open",
        "value": 2,
      },
      {
        "label": "Pending",
        "value": 3,
      },
      {
        "label": "Resolved",
        "value": 4,
      },
      {
        "label": "Closed",
        "value": 5,
      },
      {
        "label": "In Backlog",
        "value": 9,
      },
      {
        "label": "Review",
        "value": 10,
      },
    ],
    "position": 9,
    "required_for_agents": true,
    "required_for_closure": false,
    "required_for_customers": false,
    "type": "default_status",
    "updated_at": "2024-11-19T04:46:55Z",
  },
  {
    "choices": {
      "High": 3,
      "Low": 1,
      "Medium": 2,
      "Urgent": 4,
    },
    "created_at": "2018-05-09T03:34:50Z",
    "customers_can_edit": true,
    "customers_can_filter": false,
    "default": true,
    "description": "Ticket priority",
    "displayed_to_customers": true,
    "id": 42000317073,
    "label": "Priority",
    "label_for_customers": "Priority",
    "name": "priority",
    "options": [
      {
        "label": "Low",
        "value": 1,
      },
      {
        "label": "Medium",
        "value": 2,
      },
      {
        "label": "High",
        "value": 3,
      },
      {
        "label": "Urgent",
        "value": 4,
      },
    ],
    "position": 10,
    "required_for_agents": true,
    "required_for_closure": true,
    "required_for_customers": false,
    "type": "default_priority",
    "updated_at": "2022-07-01T03:49:26Z",
  },
  {
    "choices": [
      "Urgent",
      "High",
      "Medium",
    ],
    "created_at": "2023-08-07T23:26:52Z",
    "customers_can_edit": false,
    "customers_can_filter": false,
    "default": false,
    "description": null,
    "displayed_to_customers": false,
    "id": 42000341055,
    "label": "Internal Priority",
    "label_for_customers": "Internal Priority",
    "name": "cf_internal_priority",
    "options": [
      {
        "label": "Urgent",
        "value": "Urgent",
      },
      {
        "label": "High",
        "value": "High",
      },
      {
        "label": "Medium",
        "value": "Medium",
      },
    ],
    "position": 11,
    "required_for_agents": false,
    "required_for_closure": false,
    "required_for_customers": false,
    "type": "custom_dropdown",
    "updated_at": "2023-08-07T23:57:57Z",
  },
  {
    "choices": {
      "Support": 42000101825,
      "Urgent Tickets": 42000103601,
    },
    "created_at": "2018-05-09T03:34:51Z",
    "customers_can_edit": false,
    "customers_can_filter": false,
    "default": true,
    "description": "Ticket group",
    "displayed_to_customers": false,
    "id": 42000317074,
    "label": "Group",
    "label_for_customers": "Group",
    "name": "group",
    "options": [
      {
        "label": "Support",
        "value": 42000101825,
      },
      {
        "label": "Urgent Tickets",
        "value": 42000103601,
      },
    ],
    "position": 12,
    "required_for_agents": false,
    "required_for_closure": false,
    "required_for_customers": false,
    "type": "default_group",
    "updated_at": "2022-08-29T00:19:20Z",
  },
  {
    "choices": {
      "Blake Lahiff": 42002289687,
      "Daniel Radakovic": 42030962623,
      "Shannan Galbraith": 42024779981,
    },
    "created_at": "2018-05-09T03:34:51Z",
    "customers_can_edit": false,
    "customers_can_filter": false,
    "default": true,
    "description": "Agent",
    "displayed_to_customers": false,
    "id": 42000317075,
    "label": "Agent",
    "label_for_customers": "Assigned to",
    "name": "agent",
    "options": [
      {
        "label": "Blake Lahiff",
        "value": 42002289687,
      },
      {
        "label": "Daniel Radakovic",
        "value": 42030962623,
      },
      {
        "label": "Shannan Galbraith",
        "value": 42024779981,
      },
    ],
    "position": 13,
    "required_for_agents": false,
    "required_for_closure": false,
    "required_for_customers": false,
    "type": "default_agent",
    "updated_at": "2023-03-28T02:37:28Z",
  },
  {
    "created_at": "2018-05-09T03:34:51Z",
    "customers_can_edit": true,
    "customers_can_filter": false,
    "default": true,
    "description": "Ticket description",
    "displayed_to_customers": true,
    "id": 42000317077,
    "label": "Description",
    "label_for_customers": "Description",
    "name": "description",
    "position": 15,
    "required_for_agents": true,
    "required_for_closure": true,
    "required_for_customers": true,
    "type": "default_description",
    "updated_at": "2020-08-03T04:26:38Z",
  },
  {
    "choices": {
      "AUS": {
        "NSW": [
          "Sydney",
          "Gosford",
        ],
        "QLD": [
          "Brisbane",
          "Gold Coast",
        ],
      },
      "Canada": {
        "Ontario": [],
      },
      "England": {},
      "USA": {
        "Kansas": [
          "Kansas City",
          "Manhattan",
        ],
        "Texas": [
          "Austin",
          "Dallas",
        ],
      },
    },
    "created_at": "2025-02-25T02:48:31Z",
    "customers_can_edit": false,
    "customers_can_filter": false,
    "default": false,
    "description": null,
    "displayed_to_customers": false,
    "id": 42000342174,
    "label": "Country",
    "label_for_customers": "Country",
    "name": "cf_country",
    "nested_ticket_fields": [
      {
        "created_at": "2025-02-25T02:48:31Z",
        "description": null,
        "id": 42000002085,
        "label": "State",
        "label_in_portal": "State",
        "level": 2,
        "name": "cf_state",
        "ticket_field_id": 42000342174,
        "updated_at": "2025-02-25T02:48:31Z",
      },
      {
        "created_at": "2025-02-25T02:48:31Z",
        "description": null,
        "id": 42000002086,
        "label": "City",
        "label_in_portal": "City",
        "level": 3,
        "name": "cf_city",
        "ticket_field_id": 42000342174,
        "updated_at": "2025-02-25T02:48:31Z",
      },
    ],
    "options": [
      {
        "label": "AUS",
        "options": [
          {
            "label": "NSW",
            "options": [
              {
                "label": "Sydney",
                "value": "Sydney",
              },
              {
                "label": "Gosford",
                "value": "Gosford",
              },
            ],
            "value": "NSW",
          },
          {
            "label": "QLD",
            "options": [
              {
                "label": "Brisbane",
                "value": "Brisbane",
              },
              {
                "label": "Gold Coast",
                "value": "Gold Coast",
              },
            ],
            "value": "QLD",
          },
        ],
        "value": "AUS",
      },
      {
        "label": "USA",
        "options": [
          {
            "label": "Texas",
            "options": [
              {
                "label": "Austin",
                "value": "Austin",
              },
              {
                "label": "Dallas",
                "value": "Dallas",
              },
            ],
            "value": "Texas",
          },
          {
            "label": "Kansas",
            "options": [
              {
                "label": "Kansas City",
                "value": "Kansas City",
              },
              {
                "label": "Manhattan",
                "value": "Manhattan",
              },
            ],
            "value": "Kansas",
          },
        ],
        "value": "USA",
      },
      {
        "label": "Canada",
        "options": [
          {
            "label": "Ontario",
            "options": [],
            "value": "Ontario",
          },
        ],
        "value": "Canada",
      },
      {
        "label": "England",
        "options": [],
        "value": "England",
      },
    ],
    "position": 16,
    "required_for_agents": false,
    "required_for_closure": false,
    "required_for_customers": false,
    "type": "nested_field",
    "updated_at": "2025-02-25T02:48:31Z",
  },
]
`;
